classes/export_webservices.php

```diff
@@ -0,0 +1,387 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Variables and methods relating to exporting and downloading of webservice formats
+ *
+ * @package   tool_wsformat
+ * @copyright 2023 Djarran Cotleanu, Zach Pregl
+ * @author    Djarran Cotleanu, Zach Pregl
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_wsformat;
+
+use core_external\external_api;
+
+use core_external\external_multiple_structure;
+use core_external\external_single_structure;
+
+/**
+ * Class for processing and exporting web service data.
+ */
+class export_webservices {
+
+    /**
+     * Stores the host address of the server.
+     *
+     * @var string
+     */
+    private $host = '';
+
+    /**
+     * An array to hold webservice objects.
+     *
+     * @var array
+     */
+    public $webservices = [];
+
+
+    /**
+     * Constructor function - assign instance variables.
+     *
+     * @param string $host
+     * @param array  $selectedwebserviceindices
+     */
+    public function __construct(string $host, array $selectedwebserviceindices) {
+        $this->host        = $host;
+        $this->webservices = $this->get_selected_webservice_objects($selectedwebserviceindices);
+    }
+
+    /**
+     * Exports data as cURL commands in a text file.
+     * Sets header to initiate download with filename and extension.
+     */
+    public function export_as_curl(): void {
+        header('Content-Disposition: attachment; filename=curl.txt');
+        header('Content-Type: application/plain');
+
+        $curlstrings = [];
+        foreach ($this->webservices as $webservice) {
+            $paramsarray = $this->get_formatted_param_array($webservice);
+
+            $curlstrings[] = 'curl ' . $this->create_request_string($webservice, $paramsarray);
+        }
+
+        foreach ($curlstrings as $curlstring) {
+            echo $curlstring . "\n" . "\n";
```

Comment:
Can we swap these "\n" to the PHP_EOL constant?

It just does context aware outputs so it would output correctly if run on windows for example, which requires a "\r\n"